### PR TITLE
Added helm3 option to chart app

### LINKED ISSUE #20: Add --helm3 flag to chart app
## Description

Add --helm3 flag to chart app

The `arkade install chart` command should have helm3 support and use it by default.

### PR DESCRIPTION
**Pull request by kadern0**
Signed-off-by: kadern0 <kaderno@gmail.com>

## Description
Added helm3 option

## Motivation and Context
Fixes #20 

## How Has This Been Tested?

Run it locally (helm3 is the default option):

``` 
./arkade install chart --repo-name stable/nginx-ingress --helm3=true
Using kubeconfig: /home/kaderno/.kube/config
Using helm3
Client: x86_64, Linux
2020/03/03 19:44:30 User dir established as: /home/kaderno/.arkade/
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈ 
VALUES values.yaml
Command: /home/kaderno/.arkade/bin/helm3/helm [upgrade --install nginx-ingress stable/nginx-ingress --namespace default --values /tmp/charts/nginx-ingress/values.yaml]
Release "nginx-ingress" has been upgraded. Happy Helming!
NAME: nginx-ingress
LAST DEPLOYED: Tue Mar  3 19:44:37 2020
NAMESPACE: default
STATUS: deployed
REVISION: 9
TEST SUITE: None
NOTES:
The nginx-ingress controller has been installed.
It may take a few minutes for the LoadBalancer IP to be available.
You can watch the status by running 'kubectl --namespace default get services -o wide -w nginx-ingress-controller'

An example Ingress that makes use of the controller:

  apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      kubernetes.io/ingress.class: nginx
    name: example
    namespace: foo
  spec:
    rules:
      - host: www.example.com
        http:
          paths:
            - backend:
                serviceName: exampleService
                servicePort: 80
              path: /
    # This section is only required if TLS is to be enabled for the Ingress
    tls:
        - hosts:
            - www.example.com
          secretName: example-tls

If TLS is enabled for the Ingress, a Secret containing the certificate and key must also be provided:

  apiVersion: v1
  kind: Secret
  metadata:
    name: example-tls
    namespace: foo
  data:
    tls.crt: <base64 encoded cert>
    tls.key: <base64 encoded key>
  type: kubernetes.io/tls
=======================================================================
chart stable/nginx-ingress installed.
=======================================================================
		
Thanks for using arkade!
```

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Hi @kadern0 thank you for working on this.

Please can you share the command you ran, as I think you may have missed that?

@aidun could you review please?

**Comment by kadern0** [comment]
Commands where:

``` ./arkade install chart --repo-name stable/nginx-ingress --helm3=true
Using kubeconfig: /home/kaderno/.kube/config
Using helm3
Client: x86_64, Linux
2020/03/04 07:46:33 User dir established as: /home/kaderno/.arkade/
Hang tight while we grab the latest from your chart repositories...
^C
kaderno@kaderno-XPS-13-9360:~/go/src/github.com/alexellis/arkade$ ./arkade install chart --repo-name stable/nginx-ingress
Using kubeconfig: /home/kaderno/.kube/config
Using helm3
Client: x86_64, Linux
2020/03/04 07:46:38 User dir established as: /home/kaderno/.arkade/
Hang tight while we grab the latest from your chart repositories...
```

**Comment by alexellis** [comment]
This is great, can you now show that the --set variable still works?

For instance install openfaas, and turn on the "--load-balancer"?

**Comment by kadern0** [comment]
I need to start my cluster for that. I'll do it after work today.

File: cmd/apps/chart_app.go [hunk 1]
```diff
@@ -35,10 +35,12 @@ before using the generic helm chart installer command.`,
 	chartCmd.Flags().String("values-file", "", "Give the values.yaml file to use from the upstream chart repo")
 	chartCmd.Flags().String("repo-name", "", "Chart name")
 	chartCmd.Flags().String("repo-url", "", "Chart repo")
+	chartCmd.Flags().Bool("helm3", true, "Use helm3, if set to false uses helm2")
 
 	chartCmd.Flags().StringArray("set", []string{}, "Set individual values in the helm chart")
 
 	chartCmd.RunE = func(command *cobra.Command, args []string) error {
+		wait, _ := command.Flags().GetBool("wait")
```
[the remark this answers is not in the document]
Comment (aidun) [reply cmd/apps/chart_app.go]: I discussed this point in Slack with @Waterdrips. Its not your fault. Seems to be a issue in the install command. So from my side everything seems to be finde.

**Comment by kadern0** [comment]
``` ./arkade install chart --repo-name stable/nginx-ingress --set controller.service.type=NodePort
Using kubeconfig: /home/kaderno/.kube/config
Using helm3
Client: x86_64, Linux
2020/03/04 18:59:25 User dir established as: /home/kaderno/.arkade/
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈ 
VALUES values.yaml
Command: /home/kaderno/.arkade/bin/helm3/helm [upgrade --install nginx-ingress stable/nginx-ingress --namespace default --values /tmp/charts/nginx-ingress/values.yaml --set controller.service.type=NodePort]
Release "nginx-ingress" has been upgraded. Happy Helming!
NAME: nginx-ingress
LAST DEPLOYED: Wed Mar  4 18:59:34 2020
NAMESPACE: default
STATUS: deployed
REVISION: 10
TEST SUITE: None
NOTES:
The nginx-ingress controller has been installed.
Get the application URL by running these commands:
  export HTTP_NODE_PORT=$(kubectl --namespace default get services -o jsonpath="{.spec.ports[0].nodePort}" nginx-ingress-controller)
  export HTTPS_NODE_PORT=$(kubectl --namespace default get services -o jsonpath="{.spec.ports[1].nodePort}" nginx-ingress-controller)
  export NODE_IP=$(kubectl --namespace default get nodes -o jsonpath="{.items[0].status.addresses[1].address}")

  echo "Visit http://$NODE_IP:$HTTP_NODE_PORT to access your application via HTTP."
  echo "Visit https://$NODE_IP:$HTTPS_NODE_PORT to access your application via HTTPS."

An example Ingress that makes use of the controller:

  apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      kubernetes.io/ingress.class: nginx
    name: example
    namespace: foo
  spec:
    rules:
      - host: www.example.com
        http:
          paths:
            - backend:
                serviceName: exampleService
                servicePort: 80
              path: /
    # This section is only required if TLS is to be enabled for the Ingress
    tls:
        - hosts:
            - www.example.com
          secretName: example-tls

If TLS is enabled for the Ingress, a Secret containing the certificate and key must also be provided:

  apiVersion: v1
  kind: Secret
  metadata:
    name: example-tls
    namespace: foo
  data:
    tls.crt: <base64 encoded cert>
    tls.key: <base64 encoded key>
  type: kubernetes.io/tls
=======================================================================
chart stable/nginx-ingress installed.
=======================================================================
		
Thanks for using arkade!
kaderno@kaderno-XPS-13-9360:~/go/src/github.com/alexellis/arkade$ kubectl get svc
NAME                            TYPE        CLUSTER-IP    EXTERNAL-IP   PORT(S)                      AGE
kubernetes                      ClusterIP   10.96.0.1     <none>        443/TCP                      11d
mariadb                         ClusterIP   10.96.0.146   <none>        3306/TCP                     3d21h
mariadb-slave                   ClusterIP   10.96.0.219   <none>        3306/TCP                     3d21h
nginx-ingress-controller        NodePort    10.96.0.98    <none>        80:30031/TCP,443:30497/TCP   23h
nginx-ingress-default-backend   ClusterIP   10.96.0.33    <none>        80/TCP                       23h
```

**Comment by kadern0** [comment]
Seems to be working although I want to run a couple tests more (once the namespace gets deleted)

```
./arkade install chart --repo-name openfaas/openfaas --namespace openfaas --set functionNamespace=openfaas-fn --set serviceType=LoadBalancer
Using kubeconfig: /home/kaderno/.kube/config
Using helm3
Client: x86_64, Linux
2020/03/04 19:28:47 User dir established as: /home/kaderno/.arkade/
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "openfaas" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈ 
VALUES values.yaml
Command: /home/kaderno/.arkade/bin/helm3/helm [upgrade --install openfaas openfaas/openfaas --namespace openfaas --values /tmp/charts/openfaas/values.yaml --set functionNamespace=openfaas-fn --set serviceType=LoadBalancer]
Release "openfaas" has been upgraded. Happy Helming!
NAME: openfaas
LAST DEPLOYED: Wed Mar  4 19:28:53 2020
NAMESPACE: openfaas
STATUS: deployed
REVISION: 2
TEST SUITE: None
NOTES:
To verify that openfaas has started, run:

  kubectl -n openfaas get deployments -l "release=openfaas, app=openfaas"
=======================================================================
chart openfaas/openfaas installed.
=======================================================================
		
Thanks for using arkade!
```

**Comment by kadern0** [comment]
I have to manually create the namespaces to make it work, other than that, it seems to be working fine:

```
 kubectl apply -f https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml
namespace/openfaas created
namespace/openfaas-fn created
kaderno@kaderno-XPS-13-9360:~/go/src/github.com/alexellis/arkade$ ./arkade install chart --repo-name openfaas/openfaas --namespace openfaas --set functionNamespace=openfaas-fn --set serviceType=LoadBalancer
Using kubeconfig: /home/kaderno/.kube/config
Using helm3
Client: x86_64, Linux
2020/03/04 21:13:14 User dir established as: /home/kaderno/.arkade/
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "openfaas" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete. ⎈ Happy Helming!⎈ 
VALUES values.yaml
Command: /home/kaderno/.arkade/bin/helm3/helm [upgrade --install openfaas openfaas/openfaas --namespace openfaas --values /tmp/charts/openfaas/values.yaml --set functionNamespace=openfaas-fn --set serviceType=LoadBalancer]
Release "openfaas" does not exist. Installing it now.
NAME: openfaas
LAST DEPLOYED: Wed Mar  4 21:13:19 2020
NAMESPACE: openfaas
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
To verify that openfaas has started, run:

  kubectl -n openfaas get deployments -l "release=openfaas, app=openfaas"
=======================================================================
chart openfaas/openfaas installed.
=======================================================================
		
Thanks for using arkade!

kubectl get svc -n openfaas
NAME                TYPE           CLUSTER-IP    EXTERNAL-IP   PORT(S)          AGE
alertmanager        ClusterIP      10.96.0.220   <none>        9093/TCP         37s
basic-auth-plugin   ClusterIP      10.96.0.155   <none>        8080/TCP         36s
gateway             ClusterIP      10.96.0.15    <none>        8080/TCP         36s
gateway-external    LoadBalancer   10.96.0.14    <pending>     8080:30301/TCP   36s
nats                ClusterIP      10.96.0.251   <none>        4222/TCP         37s
prometheus          ClusterIP      10.96.0.68    <none>        9090/TCP         37s

```

**Comment by kadern0** [comment]
@Waterdrips, you are right, the --wait option doesn't work, but it is defined exactly same way as with other apps (istio, mongodb, nginx...). I've tried these and I'm getting same error across all of them:
```
./arkade install chart --repo-name stable/nginx-ingress --set controller.service.type=NodePort --wait=true
Error: unknown flag: --wait
 ./arkade install mongodb --wait=true
Error: unknown flag: --wait
 ./arkade install mongodb --wait
Error: unknown flag: --wait
 ./arkade install nginx --wait
Error: unknown flag: --wait
Usage:
  arkade install [flags]
  arkade install [command]

Examples:
  arkade install
  arkade install openfaas --helm3 --gateways=2
  arkade install inlets-operator --token-file $HOME/do-token

Available Commands:
  cert-manager            Install cert-manager
  chart                   Install the specified helm chart
  cron-connector          Install cron-connector for OpenFaaS
  crossplane              Install Crossplane
  docker-registry         Install a Docker registry
  docker-registry-ingress Install registry ingress with TLS
  info                    Find info about a Kubernetes app
  inlets-operator         Install inlets-operator
  istio                   Install istio
  kafka-connector         Install kafka-connector for OpenFaaS
  kubernetes-dashboard    Install kubernetes-dashboard
  linkerd                 Install linkerd
  metrics-server          Install metrics-server
  minio                   Install minio
  mongodb                 Install mongodb
  nginx-ingress           Install nginx-ingress
  openfaas                Install openfaas
  openfaas-ingress        Install openfaas ingress with TLS
  postgresql              Install postgresql

Flags:
  -h, --help                help for install
      --kubeconfig string   Local path for your kubeconfig file (default "kubeconfig")

Use "arkade install [command] --help" for more information about a command.
```

Also I see inconsistencies printing the usage message (some apps don't do it). @alexellis could you throw some light here?

**Comment by alexellis** [comment]
Hi @kadern0 thanks for testing your work. I haven't merged this yet, because it seemed like you and @Waterdrips are in the middle of debugging an issue with `--wait`?

Are you asking me to merge with the issue in place, or did you want to spend more time on this?

**Comment by kadern0** [comment]
I'm wondering whether I'm not running the commands properly (the --wait option) or just they don't work for any of the apps. Any idea why the option is failing?

**Comment by Waterdrips** [comment]
@kadern0 

https://github.com/alexellis/arkade/blob/master/cmd/install.go#L31

This is where its defined on master. You will need to rebase your branch and then re-test with and without --wait.

**Comment by alexellis** [comment]
Please don't push a merge commit into this PR.

Assuming you have a remote for upstream, something like this would work:

```
git remote add upstream https://github.com/alexellis/arkade
git pull upstream/master --rebase
```

**Comment by kadern0** [comment]
@Waterdrips I think it is working now. Could you give it a go?

**Comment by kadern0** [comment]
Any update on this?

**Comment by alexellis** [comment]
FYI we are removing the helm3 flag so that it's always helm3. Thank you for the work on the PR. We do have plenty more opportunities to contribute. Maybe see you on Slack?

**Pull request closed without merging**